recordset/recordset.js

```diff
@@ -127,6 +127,14 @@
         $scope.unfiltered = function () {
             return recordsetModel.reference.unfilteredReference.contextualize.compact.appLink;
         };
+        
+        $scope.csvLink = function () {
+            // before run, use window location
+            if (!recordsetModel.reference) {
```

Comment:
I don't think this condition is necessary. You should only use the reference function `csvDownloadLink` like you have on line 136.

Even more so, this function is unnecessary. You can just define the link directly in the HTML. See my comment above.